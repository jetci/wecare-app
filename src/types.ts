// Common type definitions

/** Notification item */
export interface Notification {
  id: string;
  message: string;
}

/** Data point for reports chart */
export interface ReportData {
  name: string;
  value: number;
}

/** User request for admin approval */
export interface UserRequest {
  id: string;
  username: string;
  email: string;
  approved: boolean;
}
