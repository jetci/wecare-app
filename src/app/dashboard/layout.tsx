"use client";

import React, { ReactNode, useEffect } from "react";
import Link from "next/link";
import { useRouter } from "next/navigation";
import useSWR from "swr";
import Header from "@/components/layout/Header";
import { Role } from "@/types/roles";

const fetcher = (url: string) =>
  fetch(url, {
    headers: { Authorization: `Bearer ${localStorage.getItem("accessToken")}` },
  }).then((res) => res.json());

const allDashboards = [
  { key: "community", title: "ประชาชน", path: "/dashboard/community" },
  { key: "driver", title: "เจ้าหน้าที่กู้ชีพ/คนขับรถ", path: "/dashboard/driver" },
  { key: "health-officer", title: "เจ้าหน้าที่สาธารณสุข", path: "/dashboard/health-officer" },
  { key: "executive", title: "ผู้บริหาร", path: "/dashboard/executive" },
  { key: "admin", title: "ผู้ดูแลระบบ", path: "/dashboard/admin" }
];

interface Props {
  children: ReactNode;
  role: string;
}

export default function DashboardLayout({ children, role }: Props) {
  const router = useRouter();
  const { data: profileRaw } = useSWR("/api/auth/profile", fetcher);
  const profile = profileRaw?.user ?? profileRaw;
  const isDevUser = profile?.nationalId === "3500200461028";

  // Sidebar items
  let sidebarItems;
  if (role === 'community') {
    // Community users see only community dashboard
    sidebarItems = allDashboards.filter((m) => m.key === role);
  } else if (isDevUser) {
    // Dev user sees all dashboards
    sidebarItems = allDashboards;
  } else {
    // Other users see their own dashboard only
    sidebarItems = allDashboards.filter((m) => m.key === role);
  }

  // inactivity logout
  useEffect(() => {
    const events = ["click", "keydown", "mousemove", "scroll"];
    const updateActivity = () =>
      localStorage.setItem("lastActivity", Date.now().toString());
    events.forEach((e) => window.addEventListener(e, updateActivity));
    const interval = setInterval(() => {
      const last = parseInt(localStorage.getItem("lastActivity") || "0", 10);
      if (Date.now() - last > 120 * 60 * 1000) router.push("/login");
    }, 60 * 1000);
    return () => {
      events.forEach((e) => window.removeEventListener(e, updateActivity));
      clearInterval(interval);
    };
  }, [router]);

  return (
    <div className="flex flex-col h-screen">
      <Header />
      <div className="flex flex-1">
        <aside className="w-64 bg-gray-800 text-white p-4">
          <h2 className="text-xl font-bold mb-4">WeCare Dev Nav</h2>
          <nav className="space-y-2">
            {sidebarItems.map((item) => (
              <Link
                key={item.key}
                href={item.path}
                className="block p-2 hover:bg-gray-700 rounded"
              >
                {item.title}
              </Link>
            ))}
          </nav>
          <button
            onClick={() => {
              localStorage.removeItem("accessToken");
              document.cookie =
                "refreshToken=; path=/; secure; SameSite=Lax; max-age=0";
              router.push("/login");
            }}
            className="block mt-4 hover:underline"
          >
            ออกจากระบบ
          </button>
        </aside>
        <main className="flex-1 overflow-auto bg-gray-100 p-6">
          {children}
        </main>
      </div>
    </div>
  );
}