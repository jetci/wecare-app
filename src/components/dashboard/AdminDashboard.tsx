import React, { useState } from 'react'
import useSWR from 'swr'
import { User, LogEntry, ApiKey, ConfirmInfo, UnknownObject } from '@/types/components'
import { ConfirmationModal } from '../ui/ConfirmationModal'
import { SummaryCard } from '../ui/SummaryCard'
import toast from 'react-hot-toast'

const fetcher = (url: string) => fetch(url).then(res => res.json())
const tabs = [
  'Users', 'Logs', 'Settings', 'API Keys', 'Maintenance', 'Thresholds'
]

const AdminDashboard: React.FC = () => {
  const [activeTab, setActiveTab] = useState<string>(tabs[0])
  const [confirm, setConfirm] = useState<ConfirmInfo | null>(null)

  // Data
  const { data: users, error: usersError, mutate: mutateUsers } = useSWR<User[]>('/api/admin/users', fetcher)
  const { data: logs, error: logsError } = useSWR<LogEntry[]>('/api/admin/logs', fetcher)
  const { data: settings, error: settingsError } = useSWR('/api/admin/settings', fetcher)
  const { data: apiKeys, error: keysError, mutate: mutateKeys } = useSWR<ApiKey[]>('/api/admin/api-keys', fetcher)
  const { data: thresholds, error: thrError, mutate: mutateThr } = useSWR<UnknownObject[]>('/api/admin/settings/thresholds', fetcher)

  // Actions
  const handleAction = async (url: string, method='PUT', body?: UnknownObject, refresh?: () => void) => {
    try {
      const res = await fetch(url, { method, headers: {'Content-Type':'application/json'}, body: body && JSON.stringify(body) })
      if (!res.ok) throw new Error('Action failed')
      toast.success('สำเร็จ')
      refresh && refresh()
    } catch (err) {
      toast.error('ล้มเหลว')
    }
  }

  return (
    <div className="p-4">
      {/* Tabs */}
      <div className="flex space-x-4 border-b mb-4">
        {tabs.map(tab => (
          <button key={tab}
            className={`px-4 py-2 ${activeTab===tab ? 'border-b-2 border-blue-500' : ''}`}
            onClick={() => setActiveTab(tab)}>{tab}</button>
        ))}
      </div>
      <div>
        {activeTab === 'Users' && (
          <div>
            {!users && <p>Loading users...</p>}
            {usersError && <p className="text-red-500">Error loading users</p>}
            {users?.map((u: User) => (
              <div key={u.id} className="flex justify-between p-2 border-b">
                <span>{u.firstName} {u.lastName} ({u.role})</span>
                <div className="space-x-2">
                  <button className="px-2 py-1 bg-green-500 text-white rounded"
                    onClick={() => handleAction(`/api/admin/users/${u.id}/approve`, 'PUT', undefined, mutateUsers)}>
                    Approve
                  </button>
                  <button className="px-2 py-1 bg-red-500 text-white rounded"
                    onClick={() => handleAction(`/api/admin/users/${u.id}/disable`, 'PUT', undefined, mutateUsers)}>
                    Disable
                  </button>
                  <button className="px-2 py-1 bg-yellow-500 text-white rounded"
                    onClick={() => handleAction(`/api/admin/users/${u.id}/reset-password`, 'POST')}>
                    Reset PW
                  </button>
                </div>
              </div>
            ))}
          </div>
        )}
        {activeTab === 'Logs' && (
          <div>
            {!logs && <p>Loading logs...</p>}
            {logsError && <p className="text-red-500">Error loading logs</p>}
            <div className="max-h-64 overflow-auto">
              {logs?.map((log: LogEntry, i: number) => (
                <div key={i} className="text-sm border-b p-1">
                  [{log.timestamp}] {log.user} {log.action} from {log.ip}
                </div>
              ))}
            </div>
          </div>
        )}
        {activeTab === 'Settings' && (
          <div>
            {/* System Settings form */}
            {!settings && <p>Loading settings...</p>}
            {settings && <pre>{JSON.stringify(settings,null,2)}</pre>}
          </div>
        )}
        {activeTab === 'API Keys' && (
          <div>
            <button className="mb-2 px-3 py-1 bg-blue-500 text-white rounded"
              onClick={() => handleAction('/api/admin/api-keys','POST',undefined, mutateKeys)}>
              Create Key
            </button>
            {keysError && <p className="text-red-500">Error</p>}
            {apiKeys?.map((k: ApiKey) => (
              <div key={k.id} className="flex justify-between p-1 border-b">
                <span>{k.key}</span>
                <button className="px-2 py-1 bg-red-500 text-white rounded"
                  onClick={() => handleAction(`/api/admin/api-keys/${k.id}`,'DELETE',undefined, mutateKeys)}>
                  Revoke
                </button>
              </div>
            ))}
          </div>
        )}
        {activeTab === 'Maintenance' && (
          <div className="space-y-2">
            <button className="px-4 py-2 bg-blue-500 text-white rounded"
              onClick={() => handleAction('/api/admin/backup','POST')}>Backup Data</button>
            <button className="px-4 py-2 bg-gray-500 text-white rounded"
              onClick={() => handleAction('/api/admin/cache/clear','POST')}>Clear Cache</button>
          </div>
        )}
        {activeTab === 'Thresholds' && (
          <div>
            {!thresholds && <p>Loading...</p>}
            {thresholds && <pre>{JSON.stringify(thresholds,null,2)}</pre>}
          </div>
        )}
      </div>
      {confirm && (
        <ConfirmationModal
          open
          title="Confirm Action"
          message={confirm.message}
          onConfirm={() => { confirm.onConfirm(); setConfirm(null) }}
          onCancel={() => setConfirm(null)}
        />
      )}
    </div>
  )
}

export default AdminDashboard
