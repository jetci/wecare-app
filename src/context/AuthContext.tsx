"use client";

import React, { createContext, useContext, useState, useEffect, ReactNode, useMemo } from 'react';
import { useRouter } from 'next/navigation';
import type { ApiProfile } from '@/types/api';
import { apiFetch } from '@/lib/apiFetch';

export type User = ApiProfile;

export interface AuthContextType {
  loading: boolean;
  user: User | null;
  role: string | null;
  isAuthenticated: boolean;
  login: (user: User) => void; // No longer needs accessToken
  logout: () => void;
  setRole: (role: string) => void;
  isAdmin: boolean;
  isGuest: boolean;
  initialChecked: boolean;
}

const defaultAuth: AuthContextType = {
  loading: true, // Always start with loading true
  user: null,
  role: null,
  isAuthenticated: false,
  login: () => {},
  logout: () => {},
  setRole: () => {},
  isAdmin: false,
  isGuest: true,
  initialChecked: false, // Start with false, so we show a loading state
};

const AuthContext = createContext<AuthContextType>(defaultAuth);

export const useAuth = () => useContext(AuthContext);

export const AuthProvider = ({ children }: { children: ReactNode }) => {
  const [user, setUser] = useState<User | null>(null);
  const [loading, setLoading] = useState(true);
  const [role, setRoleState] = useState<string | null>(null);
  const [isAuthenticated, setIsAuthenticated] = useState(false);
  const [initialChecked, setInitialChecked] = useState(false);
  const router = useRouter();

  useEffect(() => {
    const fetchUserProfile = async () => {
      console.log('[AuthContext] Attempting to fetch profile via cookie...');
      try {
        const response = await apiFetch('/api/auth/profile');
        if (!response.ok) {
          throw new Error('Failed to fetch profile');
        }

        const data = await response.json();
        // The actual user data might be nested inside a 'user' property
        const profile: User = data.user || data;

        console.log('[AuthContext] Profile fetched successfully:', profile);
        setUser(profile);
        setRoleState(profile.role);
        setIsAuthenticated(true);
      } catch (error) {
        console.log('[AuthContext] No active session found or error fetching profile.');
        setUser(null);
        setRoleState(null);
        setIsAuthenticated(false);
      } finally {
        setLoading(false);
        setInitialChecked(true); // Mark initial check as complete
      }
    };

    fetchUserProfile();
  }, []); // Run only once on initial client-side mount

  const login = (userData: User) => {
    console.log('[AuthContext] Login successful, updating state and redirecting.');
    // The HttpOnly cookie is already set by the /api/login endpoint.
    // We just need to update the client-side state and redirect.
    setUser(userData);
    setRoleState(userData.role);
    setIsAuthenticated(true);

    const roleToPath: Record<string, string> = {
      COMMUNITY: '/dashboard/community',
      DRIVER: '/dashboard/driver',
      OFFICER: '/dashboard/health-officer',
      EXECUTIVE: '/dashboard/executive',
      ADMIN: '/dashboard/admin',
      DEVELOPER: '/dashboard/developer',
    };
    const redirectPath = roleToPath[userData.role.toUpperCase()] || '/dashboard/community';
    router.push(redirectPath);
  };

  const logout = async () => {
    console.log('[AuthContext] Logout function called.');
    try {
      // Call the backend to clear the HttpOnly cookie.
      // This is more secure than just clearing client-side state.
      await apiFetch('/api/logout', { method: 'POST' });
    } catch (error) {
      console.error('[AuthContext] Error calling logout API:', error);
    } finally {
      // Clear client state regardless of API call success and redirect.
      setUser(null);
      setRoleState(null);
      setIsAuthenticated(false);
      router.push('/login');
    }
  };

  const setRole = (newRole: string) => {
    setRoleState(newRole);
    setUser(prev => (prev ? { ...prev, role: newRole } : null));
  };

  const isAdmin = useMemo(() => role === 'ADMIN', [role]);
  const isGuest = useMemo(() => !isAuthenticated, [isAuthenticated]);

  const authValue = useMemo(() => ({
    user,
    role,
    isAuthenticated,
    loading,
    login,
    logout,
    setRole,
    isAdmin,
    isGuest,
    initialChecked,
  }), [user, role, isAuthenticated, loading, isAdmin, isGuest, initialChecked]);

  // This guard clause is crucial for preventing hydration mismatch.
  // It ensures that the children are not rendered on the client until the initial auth check is complete.
  if (!initialChecked) {
    return (
      <div suppressHydrationWarning={true} className="flex items-center justify-center min-h-screen w-full">
        <p>Initializing Authentication...</p>
      </div>
    );
  }

  return (
    <AuthContext.Provider value={authValue}>
      {children}
    </AuthContext.Provider>
  );
};

export default AuthContext;
