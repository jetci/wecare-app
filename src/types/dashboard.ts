// src/types/dashboard.ts

import type { ComponentType } from 'react';

export interface CacheStats {
  hits: number
  misses: number
  keysCount: number
}

export interface JobStats {
  successCount: number
  failedCount: number
  items: unknown[] // TODO: define Job type
}

export interface DeploymentStats {
  pendingCount: number
  successCount: number
  failureCount: number
  lastDeployed: string // ISO timestamp
}

export interface HealthPoint {
  timestamp: string // ISO timestamp
  cpuUsage: number
  memoryUsage: number
  dbConnections: number
}

export interface SystemHealth {
  cpuUsage: number
  memoryUsage: number
  dbConnections: number
}

export interface Notification {
  id: string
  message: string
  timestamp: string // ISO timestamp
}

export interface Community {
  id: string
  name: string
  total: number
  inCare: number
  transferred: number
}

export interface Patient {
  id: string
  name: string
  status: string
}

// Dashboard menu item type
export interface DashboardMenuItem {
  key: string
  title: string
  path: string
  icon?: ComponentType<unknown>
  children?: DashboardMenuItem[]
}

export interface DashboardMenus {
  community: DashboardMenuItem[]
  driver: DashboardMenuItem[]
  healthOfficer: DashboardMenuItem[]
  executive: DashboardMenuItem[]
  admin: DashboardMenuItem[]
}

// Menu structure configuration placeholder
// define dashboardMenus in a separate file

export interface Patient {
  id: string
  name: string
  status: string
}