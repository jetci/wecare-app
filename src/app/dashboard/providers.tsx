'use client';
import { AuthProvider } from '@/context/AuthContext';

export function DashboardProviders({ children }: { children: React.ReactNode }) {
  return <AuthProvider>{children}</AuthProvider>;
}
