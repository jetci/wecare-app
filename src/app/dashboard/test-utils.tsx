import React, { type ReactNode } from 'react';
import {
  render as rtlRender,
  screen,
  fireEvent,
  waitFor,
  act,
  within,
  waitForElementToBeRemoved,
  cleanup,
  type RenderOptions,
} from '@testing-library/react';
// Removed renderHook import; hook testing should use render from @testing-library/react or custom wrappers
import { vi } from 'vitest';
import RoleGuard from '@/components/RoleGuard';
import { AuthProvider } from '@/context/AuthContext';
import { Role } from '@/types/roles';

// Mock next/navigation
vi.mock('next/navigation', () => ({
  useRouter: () => ({
    push: (to: string) => window.history.pushState({}, '', to),
    replace: (to: string) => window.history.pushState({}, '', to),
    prefetch: vi.fn(),
    refresh: vi.fn(),
    asPath: window.location.pathname,
  }),
  usePathname: () => window.location.pathname,
  useSearchParams: () => new URLSearchParams(window.location.search),
}));

// Mock Google Maps API
vi.mock('@react-google-maps/api', () => ({
  GoogleMap: (props: any) => <div data-testid="mock-google-map" {...props} />,
  Marker: (props: any) => <div data-testid="mock-google-marker" {...props} />,
  useJsApiLoader: () => ({ isLoaded: true, loadError: null }),
}));

interface ProvidersProps {
  children: ReactNode;
  route?: string;
  role?: string;
}

function AllProviders({ children, route = '/', role = 'ADMIN' }: ProvidersProps) {
  window.history.pushState({}, '', route);
  const roleEnum = (Role as any)[role.toUpperCase() as keyof typeof Role] as Role;
  return (
    <AuthProvider>
      <RoleGuard allowedRoles={[roleEnum]}>{children}</RoleGuard>
    </AuthProvider>
  );
}

function renderWithProviders(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'> & { route?: string; role?: string }
) {
  return rtlRender(ui, {
    wrapper: ({ children }) => <AllProviders {...options}>{children}</AllProviders>,
  });
}

export {
  renderWithProviders,
  renderWithProviders as render,
  screen,
  fireEvent,
  waitFor,
  act,
  within,
  waitForElementToBeRemoved,
  cleanup,
  // renderHook removed,
};
