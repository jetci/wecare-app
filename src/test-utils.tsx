// src/test-utils.tsx
import React from 'react'
import { render as rtlRender, RenderOptions } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { AuthProvider } from '@/context/AuthContext'
import RoleGuard from '@/components/RoleGuard'
import { Role } from '@/types/roles'

export function render(ui: React.ReactElement, options?: RenderOptions) {
  return rtlRender(
    <AuthProvider>
      <RoleGuard allowedRoles={[Role.COMMUNITY, Role.ADMIN, Role.DEVELOPER]}>{ui}</RoleGuard>
    </AuthProvider>,
    options
  )
}

export * from '@testing-library/react'
export { userEvent }
