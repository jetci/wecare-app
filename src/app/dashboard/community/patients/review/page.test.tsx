import React from 'react'
import { CacheStats, JobStats, DeploymentStats, SystemHealth, HealthPoint, Notification, Community, Patient } from '@/types/dashboard'
import { render, screen, fireEvent, waitFor, act, within, waitForElementToBeRemoved, cleanup, renderHook } from '@/app/dashboard/test-utils'
import { vi } from 'vitest'
// Mock Next.js useRouter with shared mockPush for redirect assertions
const mockPush = vi.fn()
vi.mock('next/navigation', async () => {
  const actual = await vi.importActual<typeof import('next/navigation')>('next/navigation')
  return {
    __esModule: true,
    ...actual,
    useRouter: () => ({
      push: mockPush,
      replace: vi.fn(),
      prefetch: vi.fn(),
      refresh: vi.fn(),
      forward: vi.fn(),
      back: vi.fn(),
      asPath: window.location.pathname + window.location.search,
      pathname: window.location.pathname,
    }),
    usePathname: () => window.location.pathname,
    useParams: () => ({}),
    useSearchParams: () => new URLSearchParams(window.location.search),
  }
})

// Mock confirmPatientData for all tests
const mockConfirm = vi.fn()
vi.mock('@/lib/confirmPatientData', () => ({
  __esModule: true,
  confirmPatientData: (...args: any[]) => mockConfirm(...args),
}))

import * as AuthContext from '@/context/AuthContext'
import PatientReviewPage from './page'
// react-hot-toast is mocked globally in vitest.setup.ts

describe('PatientReviewPage', () => {
  beforeEach(() => {
    vi.clearAllMocks()
    mockConfirm.mockReset()
    window.alert = vi.fn()
    global.fetch = vi.fn().mockResolvedValue({ ok: true, text: async () => '' })
  })

  it('renders patient data from params', () => {
    const route = '/dashboard/community/patients/review?prefix=นาย&firstName=Somchai&lastName=Sukjai'
    const { container } = render(<PatientReviewPage />, { role: 'COMMUNITY', route })
    expect(screen.getByText(/Somchai/)).toBeInTheDocument()
    expect(screen.getByText(/Sukjai/)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('alerts when no patientGroup and prevents submission', async () => {
    const route = '/dashboard/community/patients/review?prefix=นาย&firstName=Somchai&lastName=Sukjai'
    render(<PatientReviewPage />, { role: 'COMMUNITY', route })
    fireEvent.click(screen.getByRole('button', { name: /ยืนยันและบันทึก/ }))
    await waitFor(() => expect(window.alert).toHaveBeenCalledWith('กรุณาเลือกกลุ่มผู้ป่วย'))
  })

  it('redirects on successful save', async () => {
    const route = '/dashboard/community/patients/review?prefix=นาย&firstName=Somchai&lastName=Sukjai&patientGroup=ผู้ป่วยติดเตียง'
    // mock confirmPatientData resolution
    mockConfirm.mockResolvedValueOnce(new Response('', { status: 200 }))
    render(<PatientReviewPage />, { role: 'COMMUNITY', route })
    await act(async () => {
      fireEvent.click(screen.getByRole('button', { name: /ยืนยันและบันทึก/ }))
    })
    console.log(mockPush.mock.calls)
    await waitFor(
      () => expect(mockPush).toHaveBeenCalledWith(expect.stringMatching(/\/dashboard\/community\/patients(\/new)?$/)),
      { timeout: 2000 }
    )
  })

  it('shows error message on failure', async () => {
    const route = '/dashboard/community/patients/review?prefix=นาย&firstName=Somchai&lastName=Sukjai&patientGroup=ผู้ป่วยติดเตียง'
    mockConfirm.mockResolvedValueOnce(new Response('Bad', { status: 500 }))
    render(<PatientReviewPage />, { role: 'COMMUNITY', route })
    fireEvent.click(screen.getByRole('button', { name: /ยืนยันและบันทึก/ }))
    expect(await screen.findByTestId('toast-error')).toBeInTheDocument()
  })

  it('redirects to new form if no params', async () => {
    const route = '/dashboard/community/patients/review'
    await act(async () => {
      render(<PatientReviewPage />, { role: 'COMMUNITY', route })
    })
    console.log(mockPush.mock.calls)
    await waitFor(
      () => expect(mockPush).toHaveBeenCalledWith(expect.stringMatching(/\/dashboard\/community\/patients(\/new)?$/)),
      { timeout: 2000 }
    )
  })

  it('redirects non-community users', async () => {
    const route = '/dashboard/community/patients/review?firstName=X'
    // Mock useAuth to return non-community role
    vi.spyOn(AuthContext, 'useAuth').mockReturnValue({
      user: null,
      role: 'ADMIN',
      isAuthenticated: true,
      setRole: vi.fn(),
      isAdmin: false,
      isGuest: false,
    })
    // Render without manual act to allow built-in effect flushing
    render(<PatientReviewPage />, { role: 'ADMIN', route })
    console.log('non-community mockPush calls:', mockPush.mock.calls)
    await waitFor(
      () => expect(mockPush).toHaveBeenCalledWith(expect.stringMatching(/\/dashboard\/community\/patients$/)),
      { timeout: 2000 }
    )
  })
})
