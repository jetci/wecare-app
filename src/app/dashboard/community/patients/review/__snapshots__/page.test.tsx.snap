// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PatientReviewPage > renders patient data from params 1`] = `
<div>
  <div
    class="max-w-2xl mx-auto p-6 bg-white rounded shadow"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      ตรวจสอบข้อมูลผู้ป่วย
    </h1>
    <div
      class="space-y-2 mb-6"
    >
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          คำนำหน้า:
        </span>
        <span>
          นาย
        </span>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          ชื่อ:
        </span>
        <span>
          Somchai
        </span>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          สกุล:
        </span>
        <span>
          Sukjai
        </span>
      </div>
      <div
        class="flex items-center space-x-2"
      >
        <span
          class="w-40 font-semibold"
        >
          รูปภาพผู้ป่วย:
        </span>
        <span
          class="text-red-500"
        >
          ไม่มีรูปภาพผู้ป่วย
        </span>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          เลขบัตรประชาชน:
        </span>
        <span />
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          เบอร์โทร:
        </span>
        <span />
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          เพศ:
        </span>
        <span />
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          กรุ๊ฟเลือด:
        </span>
        <span />
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          วันเกิด:
        </span>
        <span>
          ไม่มีข้อมูล
        </span>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          อายุ:
        </span>
        <span />
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          ที่อยู่:
        </span>
        <span>
          เลขที่ undefined หมู่ที่ undefined หมู่บ้าน undefined ตำบล เวียง อำเภอ ฝาง จังหวัด เชียงใหม่
        </span>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          สถานะการช่วยเหลือตัวเอง:
        </span>
        <span />
      </div>
      <div
        class="space-y-2 mb-4"
      >
        <div
          class="flex"
        >
          <span
            class="w-40 font-semibold"
          >
            หนังสือรับรอง กำนัน/ผู้ใหญ่บ้าน:
          </span>
          <span>
            ไม่มีเอกสารแนบ
          </span>
        </div>
        <div
          class="flex"
        >
          <span
            class="w-40 font-semibold"
          >
            หนังสือรับรองผู้ป่วยติดเตียง:
          </span>
          <span>
            ไม่มีเอกสารแนบ
          </span>
        </div>
        <div
          class="flex"
        >
          <span
            class="w-40 font-semibold"
          >
            สำเนาใบนัดแพทย์:
          </span>
          <span>
            ไม่มีเอกสารแนบ
          </span>
        </div>
        <div
          class="flex"
        >
          <span
            class="w-40 font-semibold"
          >
            เอกสารอื่นๆ:
          </span>
          <span>
            ไม่มีเอกสารแนบ
          </span>
        </div>
      </div>
      <div
        class="flex"
      >
        <span
          class="w-40 font-semibold"
        >
          พิกัดรับส่ง:
        </span>
        <span>
          พิกัดรับส่ง: undefined, undefined
        </span>
      </div>
    </div>
    <div
      class="flex justify-end space-x-4"
    >
      <button
        class="px-4 py-2 border rounded text-gray-700"
      >
        แก้ไข
      </button>
      <button
        class="px-4 py-2 bg-indigo-600 text-white rounded"
      >
        ยืนยันและบันทึก
      </button>
    </div>
  </div>
</div>
`;
